tools/idf_monitor.py

```diff
@@ -153,13 +171,20 @@ def run(self):
                     c = self.console.getkey()
                 except KeyboardInterrupt:
                     c = '\x03'
+                except IOError:
+                    if not self.alive:
+                        # expected during _cancel when killed
+                        break
```

Comment:
This code swallows the IOError if self.alive is true. Does it make more sense to re-raise it?